xdsl/transforms/individual_rewrite.py

```diff
@@ -44,43 +44,26 @@ def match_and_rewrite(self, op: arith.DivUIOp, rewriter: PatternRewriter) -> Non
             rewriter.replace_matched_op([], [mul_op.lhs])
 
 
-INDIVIDUAL_REWRITE_PATTERNS_BY_OP_CLASS: dict[
-    type[Operation], tuple[RewritePattern, ...]
-] = {
-    arith.AddiOp: (AdditionOfSameVariablesToMultiplyByTwo(),),
-    arith.DivUIOp: (DivisionOfSameVariableToOne(),),
+INDIVIDUAL_REWRITE_PATTERNS_BY_NAME: dict[str, dict[str, RewritePattern]] = {
```

Comment:
I don't understand why this change fits into this PR? It's also unclear to me that matching on a string is better than matching on a type

Reply:
Further, why change this to not being a tuple?

Reply:
On further thought, why is this necessary at all. Can these two just be made into canonicalization patterns?

Reply:
It's at least not a regression